### ISSUE TITLE
add rockspec to repo

### ISSUE THREAD

**Issue by teto** [tami5/sqlite.lua#135]
sqlite.lua broke recently on nixpkgs. This prompted me to look into running sqlite.lua tests in nixpkgs.
I've been working for the past year on automated  neovim plugin dependency management for the past year using rockspecs to handle neovim plugin dependencies. So I was happy to see the package on luarocks.org. I wanted to try to generate the rockspec but the 
minimal_test.vim has 
```
set rtp+=.
set rtp+=../plenary.nvim/
set rtp+=../tree-sitter-lua/

runtime! plugin/plenary.vim
```
and these folders are not present on my system. Nevermind I ran the command otherwise. 
I wonder if you could support luarocks test as described here https://github.com/luvit/luv/issues/592 ? plenary is just used for testing right ? if plenary appeared in the rockspec test inputs,  nixpkgs scripts could automatically run the tests.

NB: I am curious do you yourself use nix ? (as I see home manager being mentioned in the README ^^)

**Comment by tami5** [comment]
Hey @teto, Yes, I use nix. 


I believe there is a luarocks for plenary. If plenary in lua path then test files can be ran with [busted](https://olivinelabs.com/busted/) command line tool

whats sort of issues did you face?

**Comment by teto** [comment]
luarocks supports test_dependencies in the rockspec (checkout https://www.lua.org/wshop18/Muhammad.pdf).
There is a plenary rockspec, I've uploaded it to luarocks some time ago.

The problem with the tests not running out of the box is:
- it makes contribution harder (there is a CI etc.)
- As a maintainer though  it is nice when projects follow some standard (like `make test`, `luarocks test`) as I dont need to dive into the intrinsicacies of every project.

To summarize, it would be nice to have a rockspec in the repo that allows to run the test

**Comment by tami5** [comment]
> luarocks supports test_dependencies in the rockspec (checkout https://www.lua.org/wshop18/Muhammad.pdf).
> 
> There is a plenary rockspec, I've uploaded it to luarocks some time ago.
> 
> 
> 
> The problem with the tests not running out of the box is:
> 
> - it makes contribution harder (there is a CI etc.)
> 
> - As a maintainer though  it is nice when projects follow some standard (like `make test`, `luarocks test`) as I dont need to dive into the intrinsicacies of every project.
> 
> 
> 
> To summarize, it would be nice to have a rockspec in the repo that allows to run the test

Definitely, I will try to ease the dev interface a bit in follow pr. Thanks

**Comment by teto** [comment]
done in https://github.com/tami5/sqlite.lua/pull/136. Do you think it's possible to do a new release https://luarocks.org/modules/tami5/sqlite/v1.2.0-0 ? I wanted to update nixpkgs and enable the tests  for sqlite.lua. 
I've got some code where nix autodetects dependencies via the rockspec, I find it pretty cool and want to put it to the test :)

**Comment by tami5** [comment]
> done in #136. Do you think it's possible to do a new release https://luarocks.org/modules/tami5/sqlite/v1.2.0-0 ? I wanted to update nixpkgs and enable the tests for sqlite.lua.
> I've got some code where nix autodetects dependencies via the rockspec, I find it pretty cool and want to put it to the test :)

exciting, here you go https://github.com/tami5/sqlite.lua/tree/v1.2.1.

**Comment by tami5** [comment]
https://luarocks.org/modules/tami5/sqlite/v1.2.1-0

**Comment by teto** [comment]
great ! I will try to update nixpkgs this evening.
